### ISSUE TITLE
✨ App Grid feature

### ISSUE THREAD

**Issue by brendomaciel** [MrKai77/Loop#580]
### Feature Request Description

Depends on #579.

The user should be able to use a Grid to re-arrange specific applications. For example: In a simple 1x1 App Grid, the user could set their browser in the left slot and let the right slot empty.

#### The user should also be able to:
- Open the Grid setup UI (similar to Lasso's) by pressing something (like Fn + Command) while focusing a window
    - If there is an App Grid or normal Grid enabled already, this will create a new Temporary App Grid (based on the existing definitions, if it's an App Grid)
    - If there is a Temporary App Grid already, the new slot should be added to this existing Temporary App Grid only if it doesn't overlap any existing slots. If that's the case, the current Temporary App Grid should be disabled (so other windows will reset) and this new slot should be added to a new Temporary App Grid
- Convert an existing Grid into an App Grid by toggling a switcher in settings
    - Maybe it can be computed automatically by checking whether there's at least one application linked to any slot. In that case, no UI would be necessary

#### Temporary App Grid:

Moving an app to slots other than the one defined in the App Grid implies in creating and enabling a new Temporary App Grid (which is a copy of current pre-defined App Grid). As result the user can re-enable the pre-defined App Grid, making the window come back to its previous position, simply by using the App Grid shortcut once instead of twice.

Closing, hiding or minimizing an application doesn't imply in creating a Temporary App Grid.

The active Temporary App Grid should be dropped by disabling it or by enabling another one.

#### Nice to have:
- Save the current App Grid (including the Temporary ones) as a new App Grid by copying it
- Re-arrange open applications automatically simply by enabling an App Grid (this behavior can be enabled/disabled in settings)
    - For example: In case there are two applications open but only one is linked to the App Grid, only this one would be resized/positioned accordingly to its slot, i.e. all the remaining applications would remain untouched
- Attach an App Grid to a specific screen
- Remember last App Grid when reconnecting a screen (in case it was enabled when the screen was disconnected)

### Final Checks

- [X] My issue is written in English
- [X] My issue title is descriptive
- [X] This is a single feature request (multiple feature requests should be reported individually)
- [X] I can help with the improvement or development of this feature (tick if you can help)
